### DISCUSSION TITLE
How to test in Nuxt 3

### DISCUSSION THREAD

**Discussion by niklasfjeldberg** [nuxt/nuxt#16240] [Questions] [answered]
The [docs](https://nuxt.com/docs/getting-started/testing) for testing says that you should install `@nuxt/test-utils-edge` and `vitest`. However when I run `nuxt test` I get the following error `[vite-node] Failed to load @nuxt/test-utils`, while I get this error `Vitest was initialized with native Node instead of Vite Node.` when i run `vitest`.

```
tests/feedback.test.js

import { describe, test, expect } from 'vitest';
import { setup, $fetch } from '@nuxt/test-utils-edge';

describe('Feedback-form', async () => {
  await setup();

  test('Submitting feedback', async () => {
    const { data } = await $fetch('/api/feedback', {
      method: 'POST',
      params: {
        emotion: 'positive',
        feedback: 'This is a test.',
        url: '/test'
      }
    });
    expect(data.value.success).toBeTruthy();
  });
});

```

What am I doing wrong?

**Comment by TheAlexLichter** [comment, marked as answer]
Please follow https://github.com/nuxt/framework/issues/2465
